noir/noir-repo/compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -70,31 +70,81 @@ pub(crate) fn optimize_into_acir(
 
     let last_array_uses = ssa.find_last_array_uses();
 
-    ssa.into_acir(brillig, abi_distinctness, &last_array_uses)
+    ssa.into_acir(&brillig, abi_distinctness, &last_array_uses)
 }
 
-/// Compiles the [`Program`] into [`ACIR`][acvm::acir::circuit::Circuit].
+/// Compiles the [`Program`] into [`ACIR``][acvm::acir::circuit::Program].
 ///
 /// The output ACIR is is backend-agnostic and so must go through a transformation pass before usage in proof generation.
 #[allow(clippy::type_complexity)]
 #[tracing::instrument(level = "trace", skip_all)]
-pub fn create_circuit(
+pub fn create_program(
     program: Program,
     enable_ssa_logging: bool,
     enable_brillig_logging: bool,
     force_brillig_output: bool,
-) -> Result<(Circuit, DebugInfo, Vec<Witness>, Vec<Witness>, Vec<SsaReport>), RuntimeError> {
+) -> Result<(AcirProgram, Vec<DebugInfo>, Vec<SsaReport>, Vec<Witness>, Vec<Witness>), RuntimeError>
```

Comment:
This is looking like it's becoming cumbersome. Need to check on whether it makes sense to have a struct for this return value or if we're doing too much.